pandas/tseries/tdi.py

```diff
@@ -440,21 +439,18 @@ def astype(self, dtype):
 
         if dtype == np.object_:
             return self.asobject
-        elif dtype == _INT64_DTYPE:
-            return self.asi8.copy()
         elif dtype == _TD_DTYPE:
             return self
         elif dtype.kind == 'm':
```

Comment:
@jreback 
And here: `dtype.kind == 'm'` --> `is_timedelta64_dtype(dtype)`?

BTW, this is the current behaviour:

```python
pd.TimedeltaIndex([1e14]).astype('m8[ns]')
Out[396]: TimedeltaIndex(['1 days 03:46:40'], dtype='timedelta64[ns]', freq=None)
>>> pd.TimedeltaIndex([1e14]).astype('m8')
Out[397]: Int64Index([100000000000000], dtype='int64')
>>> pd.TimedeltaIndex([1e14]).astype('m8[h]')
Out[398]: Int64Index([27], dtype='int64')
pd.TimedeltaIndex([1e14, np.NAN]).astype('m8[h]')
Out[399]: Float64Index([27.0, nan], dtype='float64')
```